### PR TITLE
Some docs for our changes to help answer queries

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**

### CODE REVIEW

File: THIS_FORK.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+-   browser: this contains [the work to build a Web Worker](https://github.com/microsoft/pyright/compare/main...microbit-foundation:pyright:browser)
+-   microbit: this contains the changes from browser and [some additional micro:bit customisations that are unlikey to be of general interest](https://github.com/microbit-foundation/pyright/compare/browser...microbit-foundation:pyright:microbit) - details below
+
+We don't have plans or the resource to maintain these for usecases other than the micro:bit Python Editor, but we're happy to discuss.
+
+## A new build for the browser
+
+We added a browser-pyright package alongside the CLI and VS Code extension packages. Similarly to the CLI, it's a Webpack build, but one that stubs out Node-only features. The build outputs a Web Worker.
+
+This is not currently published anywhere. If you want to use it then you'll will need to build it yourself.
```
Comment (microbit-robert) [inline THIS_FORK.md]:
```suggestion
This is not currently published anywhere. If you want to use it then you will need to build it yourself.
```

File: THIS_FORK.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Like CLI and VS Code packages, browser-pyright depends on the internals package (which contains nearly all Pyright's code).
+
+The browser package copies the high-level class from the language protocol server and configures it to run in a Web Worker environment.
+
+We've prioritised keeping the fork maintainable by minimising change over clean code.
+
+## Changes
+
+The key change is to replace the file system abstraction Pyright uses to read from disk. We took a very simple approach to this, simply swapping in the in-memory test file system class that is used in Pyrights's test suite. We then added some custom messages to allow us to manipulate that file system. This whole mechanism might be worth revisiting and comparing against tsserver's solution.
```
Comment (microbit-robert) [inline THIS_FORK.md]:
```suggestion
The key change is to replace the file system abstraction Pyright uses to read from disk. We took a very simple approach to this, simply swapping in the in-memory test file system class that is used in Pyright's test suite. We then added some custom messages to allow us to manipulate that file system. This whole mechanism might be worth revisiting and comparing against tsserver's solution.
```

File: THIS_FORK.md (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply THIS_FORK.md]: Thanks, fixed and added instructions in 18ce5e763a431f1be3285943a5a77cda2a6bbeea

File: THIS_FORK.md (hunk 2 above)
Comment (microbit-matt-hillsdon) [reply THIS_FORK.md]: e8a14f3fe9a383479873a79136dad41bfd9916f5

**Review by microbit-robert — APPROVED** [review]
One "web worker" not capitalised on line 65, otherwise, looks good.

**Pull request merged**